Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs

```
@@ -156,9 +157,22 @@ public int GatewayModeMaxConnectionLimit
         /// Gets the request timeout in seconds when connecting to the Azure Cosmos DB service.
         /// The number specifies the time to wait for response to come back from network peer.
         /// </summary>
-        /// <value>Default value is 1 minute.</value>
+        /// <value>Default value is 10 seconds.</value>
         /// <seealso cref="CosmosClientBuilder.WithRequestTimeout(TimeSpan)"/>
-        public TimeSpan RequestTimeout { get; set; }
+        public TimeSpan RequestTimeout
+        {
+            get => this.requestTimeout;
+
+            set
+            {
+                if (value.Seconds < 0)
+                {
+                    throw new ArgumentOutOfRangeException($"{nameof(this.RequestTimeout)} can not be set less than 1 second. This must be set in the Timespan of seconds.");
```

Check and error message are out of sync - code only would throw with negative timespan. error message indicates it would throw even when seconds == 0

Also this is a breaking change - throwing Argument Exceptions where it didn't before. If it is scoped to just negative timeouts it owuld be good to explain what is happening before this PR and explain why it is ok to break. If the plan is to throw any exception for postive Tiemspan i think this breaking change is unacceptable.

---

The SDK did throw an exception from version 3.0 to 3.6. The check got removed because of some changes in the direct layer. This is not a breaking change. It's simply adding back the original check.

---

It is still a breaking change if there is any minor version in the same 3.* major version that did not throw. Removing exceptions from a public API is a one-way-street, you can't revert that decision without moving to a new major version.

---

How is adding validation for an invalid value a breaking change? Throwing an invalid argument exception on a config value should be expected. It will be thrown on the set so it is a fast failure with a clear error. If the customer does any testing or validation, it will fail and they can actually fix their application. This seems to be a far better user experience then ignoring the value which will eventually lead to a live site issue because they built their app on the assumption, they could do a request timeout of 100ms.

---

@FabianMeiswinkel , @j82w any further suggestions here  or proceed with this change.

---

Semantically what @FabianMeiswinkel said is true. Post 3.6 the customers might have used values which are invalid but silently ignored now so it might break such customers. 

Its better to be very explicit on what is not supported. 
How about documenting it clearly in release notes as possible impact?

---

I am against this breaking change - what is the motivation for arguing that this breaking change is worth violating semantic versioning? How many CRIs have we seen because customers were confused that their timeouts < 1 second haven't been accepted. And if they are confused they won't like either that they can't set those shorter timeouts. So, either you open the flood gate and allow customers to set whatever timeout you want as real TimeSpan  (taking the risk that they shoot themselves in the foot and end-up with bad availability) or you keep the current behavior and document it as clearly as possible in API doc - but this breaking change is not helping much and is still causing the pain breaking changes can have.

---

Closing this PR unmerged is better than taking this change IMO. At least in V3

---

I would tend to round up or down rather than throw an exception as config wiring like this will often not traverse exception paths, much less be covered in tests IRL.
>`This must be set in the Timespan of seconds.`
If it is to proceed, perhaps a better wording is `The Timespan must be in whole Seconds.`

---

There is consensus on not introducing the breaking change. 
Let's abandon this PR.